File: test/9b45633a1769c5/python/collatz.py
def collatz(i: int):
    cnt = 0
    while i != 1:
        cnt += 1
        if i % 2 == 0:
            i /= 2
        else:
            i *= 3
            i += 1
    return cnt


for i in range(1, 1000000000000000000):
    print(collatz(i))

File: test/9b45633a1769c5/cpp/collatz.cpp
#include <iostream>
using ll = long long;

ll collatz(ll i)
{
    ll cnt = 0;
    while (i != 1)
    {
        cnt++;
        if (i % 2 == 0)
        {
            i /= 2;
        }
        else
        {
            i *= 3;
            i += 1;
        }
    }
    return cnt;
}
int main()
{
    for (ll i = 0; i < 100'0000'0000'0000'0000; i++)
    {
        std::cout << collatz(i) << std::endl;
    }
    return 0;
}